tap_redshift/__init__.py

```diff
@@ -193,8 +192,21 @@ def schema_for_column(c):
 def create_column_metadata(cols):
     mdata = metadata.new()
     mdata = metadata.write(mdata, (), 'selected-by-default', False)
+
     for c in cols:
+        valid_rep_keys = [c['name'] for c in cols
```

Comment:
You have a loop (list comprehension) inside a loop (for) here. That doesn't sound right or necessary.